src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,136 @@
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+
+            verifyAffectedRows(statement.executeUpdate());
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add book " + book.getTitle() + "to the DB", e);
+        }
+
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(SQL_FIND_BY_ID)) {
+            statement.setLong(1, id);
+
+            ResultSet resultSet = statement.executeQuery();
+            Book book = parseBook(resultSet);
+
+            return Optional.ofNullable(book);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find a book with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(SQL_FIND_ALL)) {
+            ResultSet allBooks = statement.executeQuery();
+
+            return getBooks(allBooks, books);
+        } catch (SQLException e) {
+            throw new DataProcessingException("DB is empty", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(SQL_UPDATE)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+
+            statement.executeUpdate();
+
+            Optional<Book> oldBook = findById(book.getId());
+
+            return oldBook.orElse(null);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book: " + book.getTitle(), e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(SQL_DELETE)) {
+            statement.setLong(1, id);
+
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete the book with id: " + id, e);
+        }
+    }
+
+    private static Book parseBook(ResultSet resultSet) throws SQLException {
+        Book book = null;
+
+        if (resultSet.next()) {
```

Comment:
You parse resultSet here and check if DB was updated. 
Let's rename method or remove check from here.